packages/atomic/src/components/common/generated-answer/atomic-citation/atomic-citation.tsx

```diff
@@ -96,6 +96,32 @@ export class AtomicCitation {
     );
   }
 
+  private generateTextFragmentUrl(
+    uri: string,
+    text?: string,
+    filetype?: string
+  ) {
+    if (filetype !== 'html' || !text) {
+      return uri;
+    }
+    const highlight = this.extractTextToHighlight(text);
+    const encodedTextFragment = encodeURIComponent(highlight)
+      .replace(/-/g, '%2D')
+      .replace(/_/g, '%5F')
+      .replace(/~/g, '%7E');
+    return `${uri}#:~:text=${encodedTextFragment}`;
+  }
+
+  private extractTextToHighlight(text: string) {
+    const SENTENCE_REGEX = /(?<!\d)(?<=^|[.!?]\s)["'\(\[]?[A-Z][^.!?]*[.!?]/g;
```

Comment:
Instead of trying to perfectly detect the first sentence of the chunk, did you consider leveraging `textStart` and `textEnd` and extract the first and last few words of the chunk?